### PR TITLE
[Plugins] Move Newmat matters into the new SofaNewmat plugin

### LINKED ISSUE #2483: Depreciate NewMat* (matrices and vector) and LULinearSolver
The newmat library is really old (discussion here https://github.com/sofa-framework/sofa/pull/2311)

This is the last thing inheriting BaseVector/BaseMatrix which is not in Sofa.Algebra (related to https://github.com/sofa-framework/sofa/pull/2311)

So the plan is to:
 - deprecate LULinearSolver which needs NewMat only (not possible to use with standard Matrix)
 - deprecate all NewMatMatrix and NewMatVector
 - (not related with algebra but) remove all usage of newmat itself here and there.

### PR DESCRIPTION
**Pull request by fredroy**
Second and last task about https://github.com/sofa-framework/sofa/issues/2483

Contains #2531

Create a new plugin SofaNewmat with 
 - newmat lib itself
 - NewmatMatrix and NewmatVector (and their tests)
 - LULinearSolver (only compatible with NewmatMatrix)
 
Add a  "compatibility" layer (tricky because SofaNewmat is optional
Update ComponentChange
Remove newmat usage in image plugin.

Fix  #2483 

Change the template of CGLinearSolver in the BeamLinearForceField because I dont see why we would need to run it with newmat matrices. It results with a very slight difference on the DOF positions

[ci-depends-on https://github.com/sofa-framework/Regression/pull/21]

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

File: modules/SofaDenseSolver/CMakeLists.txt [hunk 1]
```diff
@@ -42,8 +33,8 @@ sofa_create_package_with_targets(
...
-if(SOFADENSESOLVER_BUILD_TESTS)
-    enable_testing()
-    add_subdirectory(${PROJECT_NAME}_test)
-endif()
\ No newline at end of file
+# cmake_dependent_option(SOFADENSESOLVER_BUILD_TESTS "Compile the automatic tests" ON "SOFA_BUILD_TESTS OR NOT DEFINED SOFA_BUILD_TESTS" OFF)
+# if(SOFADENSESOLVER_BUILD_TESTS)
+#     enable_testing()
+#     add_subdirectory(${PROJECT_NAME}_test)
+# endif()
```
Comment (guparan) [inline modules/SofaDenseSolver/CMakeLists.txt]: If empty then remove

File: modules/SofaMiscFem/CMakeLists.txt [hunk 2]
```diff
@@ -28,11 +28,17 @@ list(APPEND HEADER_FILES
...
     ${SOFAMISCFEM_SRC}/TetrahedronHyperelasticityFEMForceField.inl
     ${SOFAMISCFEM_SRC}/TriangleFEMForceField.h
     ${SOFAMISCFEM_SRC}/TriangleFEMForceField.inl
+    ${SOFAMISCFEM_SRC}/PlasticMaterial.h
+    ${SOFAMISCFEM_SRC}/TriangularAnisotropicFEMForceField.h
+    ${SOFAMISCFEM_SRC}/TriangularAnisotropicFEMForceField.inl
+    ${SOFAMISCFEM_SRC}/TriangularFEMForceField.h
+    ${SOFAMISCFEM_SRC}/TriangularFEMForceField.inl
+	${SOFAMISCFEM_SRC}/QuadBendingFEMForceField.h
+	${SOFAMISCFEM_SRC}/QuadBendingFEMForceField.inl
```
Comment (guparan) [inline modules/SofaMiscFem/CMakeLists.txt]:
```suggestion
    ${SOFAMISCFEM_SRC}/QuadBendingFEMForceField.h
    ${SOFAMISCFEM_SRC}/QuadBendingFEMForceField.inl
```

File: modules/SofaMiscFem/CMakeLists.txt [hunk 3]
```diff
@@ -41,36 +47,18 @@ list(APPEND SOURCE_FILES
 
     ${SOFAMISCFEM_SRC}/TetrahedronHyperelasticityFEMForceField.cpp
     ${SOFAMISCFEM_SRC}/TriangleFEMForceField.cpp
+    ${SOFAMISCFEM_SRC}/PlasticMaterial.cpp
+    ${SOFAMISCFEM_SRC}/TriangularAnisotropicFEMForceField.cpp
+    ${SOFAMISCFEM_SRC}/TriangularFEMForceField.cpp
+	${SOFAMISCFEM_SRC}/QuadBendingFEMForceField.cpp
```
Comment (guparan) [inline modules/SofaMiscFem/CMakeLists.txt]:
```suggestion
    ${SOFAMISCFEM_SRC}/QuadBendingFEMForceField.cpp
```

File: applications/plugins/SofaNewmat/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    PACKAGE_VERSION ${Sofa_VERSION}
+    TARGETS ${PROJECT_NAME} AUTO_SET_TARGET_PROPERTIES
+    INCLUDE_SOURCE_DIR "src"
+    INCLUDE_INSTALL_DIR "${PROJECT_NAME}"
+    RELOCATABLE "plugins"
+)
+
+cmake_dependent_option(SOFANEWMAT_BUILD_TESTS "Compile the automatic tests" ON "SOFA_BUILD_TESTS OR NOT DEFINED SOFA_BUILD_TESTS" OFF)
+if(SOFANEWMAT_BUILD_TESTS)
+    enable_testing()
```
Comment (guparan) [inline applications/plugins/SofaNewmat/CMakeLists.txt]:
```suggestion

```

File: applications/plugins/SofaNewmat/SofaNewmat_test/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+set(SOURCE_FILES
+    NewMatMatrix_test.cpp
+)
+
+add_executable(${PROJECT_NAME} ${SOURCE_FILES})
+
+target_link_libraries(${PROJECT_NAME} Sofa.Testing SofaNewmat Sofa.LinearAlgebra.Testing)
+
+add_test(NAME ${PROJECT_NAME} COMMAND ${PROJECT_NAME})
```
Comment (guparan) [inline applications/plugins/SofaNewmat/SofaNewmat_test/CMakeLists.txt]:
```suggestion
```

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by guparan** [comment]
[ci-build]

**Pull request merged**